### PR TITLE
[backbone] wait for Backbone network interface to be ready

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit enhances `otbr-agent` to tolerant temporary Backbone interface unavailabity (especially when the device is just booted).

### CODE REVIEW

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

File: src/ncp/ncp_openthread.hpp [hunk 1]
```diff
@@ -164,6 +164,8 @@ class ControllerOpenThread : public MainloopProcessor
     ~ControllerOpenThread(void) override;
 
 private:
+    using Clock = std::chrono::steady_clock;
```
Comment (wgtdkp) [inline src/ncp/ncp_openthread.hpp:167]: Please reuse the definition in `src/common/time.hpp`, reusing the same pre-definition makes sure that we don't use `std::chrono::system_clock` for a Clock usage.

**Review by wgtdkp — APPROVED** [review]

File: src/ncp/ncp_openthread.cpp [hunk 2]
```diff
@@ -142,6 +146,31 @@ void ControllerOpenThread::Init(void)
...
+        auto deadline = Clock::now() + kTimeout;
+
+        while ((interfaceIndex = if_nametoindex(aBackboneInterfaceName)) == 0 && Clock::now() < deadline)
+        {
+            otbrLogWarning("Backbone interface '%s' is not ready, will re-check after %d seconds.",
+                           aBackboneInterfaceName, kRecheckIntervalSecs);
+            std::this_thread::sleep_for(std::chrono::seconds(kRecheckIntervalSecs));
+        }
+
+        VerifyOrDie(interfaceIndex > 0, "Backbone interface is invalid");
```
[the remark this answers is not in the document]
Comment (simonlingoogle) [reply src/ncp/ncp_openthread.cpp:209]: It prints WARN every 3 seconds. So the overall delay can be estimated from the number of WARN messages.

File: src/ncp/ncp_openthread.hpp (hunk 1 above)
Comment (simonlingoogle) [reply src/ncp/ncp_openthread.hpp]: Done

**Comment by Irving-cl** [comment]
I think we don't need this PR anymore? @simonlingoogle

**Comment by simonlingoogle** [comment]
PR no longer needed

**Pull request closed without merging**